lib/etcdec.cxx

```diff
@@ -173,8 +173,9 @@ submitted to the exclusive jurisdiction of the Swedish Courts.
 // the source of the functions or the compiled binary code.
 #if defined(_MSC_VER)
   #pragma warning(push)
-  #pragma warning(disable: 4100 4244 )
-#elif __clang__
+  #pragma warning(disable: 4100 4244)
+#endif
+#if __clang__
   #pragma clang diagnostic push
   #pragma clang diagnostic ignored "-Wunused-parameter"
   #if __clang_major__ > 13 || (__clang_major__ == 13 && __clang_minor__ >= 1)
```

Comment:
We're still getting `unused-but-set-variable` warnings in the CLangCL builds. Probably the version number here needs adjusting. I think I must have found that version 13.0.0 complained that `-Wunused-but-set-variable` was unrecognized hence the carve out.

It appears that Apple Clang has different version numbers from Clang which is why CMake inrtroduced `AppleCLang` as a compiler id. Looking at [this chart](https://en.wikipedia.org/wiki/Xcode#Toolchain_versions) Apple Clang version 13.0.0 is based on clang 12.0.0. Not sure then why I got unused-but-set as unrecognized for 13.0.0 since we're seeing it in clang 12.0.0.

Anyway we need to fix version number handling here and in CMakeLists to account for the differences. According to the chart, Xcode 14 will have Apple clang 14.0.0 which is base on clang 14.0.0 so maybe they'll stay in better sync from here on.

Ahh! The fifth answer to this [SO question](https://stackoverflow.com/questions/33603027/get-apple-clang-version-and-corresponding-upstream-llvm-version) says that in Clang you can use statements like `#if __has_warning("-Wunused-but-set-variable")`. That will be much better than version number heuristics. Please change this accordingly.

Not sure what to do in CMakeLists.